packages/react-cosmos-component-playground/src/components/FixtureList/index.js

```diff
@@ -35,9 +54,18 @@ export default class FixtureList extends Component {
 
   componentWillReceiveProps(nextProps) {
     if (
-      JSON.stringify(nextProps.fixtures) !== JSON.stringify(this.props.fixtures)
+      JSON.stringify(nextProps.fixtures) !==
+        JSON.stringify(this.props.fixtures) ||
+      nextProps.projectKey !== this.props.projectKey
     ) {
-      this.setState({ fixtureTree: fixturesToTreeData(nextProps.fixtures) });
+      getSavedExpansionState(nextProps.projectKey).then(savedExpansionState => {
+        this.setState({
+          fixtureTree: fixturesToTreeData(
+            nextProps.fixtures,
+            savedExpansionState || {}
+          )
+        });
+      });
```

Comment:
Wouldn't it make sense to prune here as well? Since fixtures can change via HMR. In which case it might make sense to bundle this logic and hide the entire expansion state business under a function like this:

```
function getFixtureTree(fixtures) {
  return getSavedExpansionState(projectKey).then(savedExpansionState => {
    const fixtureTree = fixturesToTreeData(fixtures, savedExpansionState);
    pruneUnusedExpansionState(projectKey, savedExpansionState, fixtureTree);

    return fixtureTree;
  });
}
```

Then use it like:

```
// Call this in didMount and willReceiveProps
// It has pruning built in
getFixtureTree(fixtures).then(fixtureTree => {
  this.setState({ fixtureTree });
});